test/transaction_watcher_service_test.ts

```diff
@@ -0,0 +1,90 @@
+import { RPCSubprovider, Web3ProviderEngine } from '@0x/subproviders';
+import { BigNumber, providerUtils } from '@0x/utils';
+import * as chai from 'chai';
+import 'mocha';
+import { Connection, Repository } from 'typeorm';
+
+import { getDBConnectionAsync } from '../src/db_connection';
+import { TransactionEntity } from '../src/entities';
+import { TransactionWatcherService } from '../src/services/transaction_watcher_service';
+import { TransactionStates } from '../src/types';
+import { utils } from '../src/utils/utils';
+
+import { TEST_RINKEBY_PRIVATE_KEY, TEST_RINKEBY_PUBLIC_ADDRESS, TEST_RINKEBY_RPC_URL } from './config';
+import { DummySigner } from './utils/dummy_signer';
+
+const { expect } = chai;
+const NUMBER_OF_RETRIES = 20;
+
+let providerEngine: Web3ProviderEngine;
+let signer: DummySigner;
+let transactionEntityRepository: Repository<TransactionEntity>;
+let txWatcher: TransactionWatcherService;
+let connection: Connection;
+
+const LOW_GAS_PRICE = 1337;
+const MID_GAS_PRICE = 4000000000;
+const HIGH_GAS_PRICE = 9000000000;
+const WAIT_DELAY_IN_MS = 5000;
+const SHORT_EXPECTED_MINE_TIME_IN_S = 15;
+
+async function waitUntilStatusAsync(
+    txHash: string,
+    status: TransactionStates,
+    repository: Repository<TransactionEntity>,
+): Promise<void> {
+    for (let i = 0; i < NUMBER_OF_RETRIES; i++) {
+        const tx = await repository.findOne(txHash);
+        if (tx !== undefined && tx.status === status) {
+            return;
+        }
+        await utils.delay(WAIT_DELAY_IN_MS);
+    }
+
+    throw new Error(`failed to grab transaction: ${txHash} in a ${status} state`);
+}
+
+describe('transaction watcher service', () => {
+    before(async () => {
+        providerEngine = new Web3ProviderEngine();
+        providerEngine.addProvider(new RPCSubprovider(TEST_RINKEBY_RPC_URL));
+        providerUtils.startProviderEngine(providerEngine);
+        connection = await getDBConnectionAsync();
+        transactionEntityRepository = connection.getRepository(TransactionEntity);
+        signer = new DummySigner(
```

Comment:
Perhaps `TestSigner` would be more appropriate? `Dummy` makes me think it won't actually sign anything.